File: DataSets/DataSet-New/LeetCodeTasks - Develop/LeetCodeTasks/3113_find-the-number-of-subarrays-where-boundary-elements-are-maximum/solutions/c/solution.c
#include <stdio.h>

long long numberOfSubarrays(int* nums, int numsSize) {
    long long count = 0;
    for (int i = 0; i < numsSize; i++) {
        int max = nums[i];
        // Look for the rightmost occurrence of nums[i]
        int j = i;
        while (j < numsSize && nums[j] == nums[i]) {
            j++;
        }
        j--; // Adjust to the last occurrence of nums[i]
        
        // Calculate number of subarrays where nums[i] is the max and it appears at both ends
        long long len = j - i + 1;
        count += len * (len + 1) / 2;
        
        i = j; // Skip to the end of current sequence of the same element
    }
    return count;
}

File: DataSets/DataSet-New/LeetCodeTasks - Develop/LeetCodeTasks/3113_find-the-number-of-subarrays-where-boundary-elements-are-maximum/solutions/python3/solution.py
from typing import List

class Solution:
    def numberOfSubarrays(self, nums: List[int]) -> int:
        count = 0
        n = len(nums)
        
        # Track the start and end indices of each unique value
        indices = {}
        
        for i, num in enumerate(nums):
            if num not in indices:
                indices[num] = [i, i]
            else:
                indices[num][1] = i
        
        # Compute the number of subarrays for each number
        for num, (start, end) in indices.items():
            if start != end:
                count += (start + 1) * (n - end)
        
        return count
